src/Doctrine/Orm/Metadata/Resource/DoctrineOrmResourceCollectionMetadataFactory.php

```diff
@@ -0,0 +1,100 @@
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Doctrine\Orm\Metadata\Resource;
+
+use ApiPlatform\Metadata\AbstractOperation;
+use ApiPlatform\Metadata\CollectionOperationInterface;
+use ApiPlatform\Metadata\DeleteOperationInterface;
+use ApiPlatform\Metadata\Resource\Factory\ResourceMetadataCollectionFactoryInterface;
+use ApiPlatform\Metadata\Resource\ResourceMetadataCollection;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+
+class DoctrineOrmResourceCollectionMetadataFactory implements ResourceMetadataCollectionFactoryInterface
+{
+    private $decorated;
+    private $managerRegistry;
+
+    public function __construct(ManagerRegistry $managerRegistry, ResourceMetadataCollectionFactoryInterface $decorated)
+    {
+        $this->decorated = $decorated;
+        $this->managerRegistry = $managerRegistry;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function create(string $resourceClass): ResourceMetadataCollection
+    {
+        $resourceMetadataCollection = $this->decorated->create($resourceClass);
+
+        foreach ($resourceMetadataCollection as $i => $resourceMetadata) {
+            $operations = $resourceMetadata->getOperations();
+
+            foreach ($resourceMetadata->getOperations() as $operationName => $operation) {
+                if (!$this->managerRegistry->getManagerForClass($operation->getClass()) instanceof EntityManagerInterface) {
+                    continue;
+                }
+
+                $operations->add($operationName, $this->addDefaults($operation));
+            }
+
+            $resourceMetadata = $resourceMetadata->withOperations($operations);
+
+            $graphQlOperations = $resourceMetadata->getGraphQlOperations();
+            foreach ($graphQlOperations as $operationName => $operation) {
+                if (!$this->managerRegistry->getManagerForClass($operation->getClass()) instanceof EntityManagerInterface) {
+                    continue;
+                }
+
+                $graphQlOperations[$operationName] = $this->addDefaults($operation);
+            }
+
+            $resourceMetadataCollection[$i] = $resourceMetadata->withGraphQlOperations($graphQlOperations);
+        }
+
+        return $resourceMetadataCollection;
+    }
+
+    private function addDefaults($operation): AbstractOperation
+    {
+        if (null === $operation->getProvider()) {
+            $operation = $operation->withProvider($this->getProvider($operation));
+        }
+
+        if (null === $operation->getProcessor()) {
+            $operation = $operation->withProcessor($this->getProcessor($operation));
+        }
+
+        return $operation;
+    }
+
+    private function getProvider(AbstractOperation $operation): string
+    {
+        if ($operation instanceof CollectionOperationInterface) {
+            return 'api_platform.doctrine.orm.state.collection_provider';
```

Comment:
We should alias these services with the FQCN, and use it here. This will remove the hard coupling with Symfony.

Reply:
I tried just aliasing and it doesn't work with tags + service locators this was a quick fix I need to use the class as id